internal/content_coding.go

```diff
@@ -1,18 +1,115 @@
 package internal
 
 import (
+	"bufio"
 	"bytes"
 	"compress/gzip"
 	"errors"
 	"io"
 )
 
+// NewStreamContentEncoder returns a writer that will encode the stream
+// according to the encoding type.
+func NewStreamContentEncoder(encoding string, w io.Writer) (io.Writer, error) {
```

Comment:
This isn't used outside of tests.  Are you planning to use it later?

Reply:
I'll remove it for now, originally was planning to use it in the socket_writer but ended up leaving it with the batching version.  Potentially we could use something like this to reduce memory usage.